### ISSUE TITLE
JSDoc @template: Generics in return type of returned function is not evaluated.

### ISSUE THREAD

**Issue by kennytm** [Microsoft/TypeScript#15177]
**TypeScript Version:**  2.2.2 (VSCode 1.11.1 on macOS 10.12.4)

**Code**

```js
// JavaScript.
/**
 * @param {T} a
 * @returns {function(): T}
 * @template T
 */
function g(a) {
    return () => a;
}

let h = g(1)();
```

**Expected behavior:** The type of `h` is inferred as `number`.

**Actual behavior:** The type of `h` is inferred as `T`...?

![screenshot_2017-04-13 21 57 20_ia6bpj](https://cloud.githubusercontent.com/assets/103023/25007801/4775335e-2094-11e7-811c-261be674e050.png)

**Comment by sandersn** [comment]
When we check whether a type parameter is in scope, the code currently only checks for Typescript type parameters, not for `@template` tags in the JSDoc. I'll have a fix up shortly.

**Comment by sandersn** [comment]
Fix is up at #16463
